### PR TITLE
Dynamic transceiver tuning support

### PR DESCRIPTION
**Pull request by dgsudharsan**
**- What I did**
Dynamic transceiver tuning support
**- How I did it**
Dynamically program pre-emphasis and other settings based on media detected in xcvrd. The various supported media are defined in media_settings.json

**- How to verify it**
Perform OIR of different media types. Dump the pre-emphasis values from hardware and verify that they got reflected

**- Description for the changelog**

Support for dynamically programming pre-emphasis

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: meta/saiserialize.cpp [hunk 1]
```diff
@@ -2968,3 +2968,21 @@ void sai_deserialize_queue_attr(
 
     sai_deserialize_enum(s, &sai_metadata_enum_sai_queue_attr_t, (int32_t&)attr);
 }
+
+void sai_deserialize_media_change_ntf(
```
Comment (lguohan) [inline meta/saiserialize.cpp]: which attribute? do we already have the serialize for the vector<uint32_t>?

File: meta/sai_serialize.h [hunk 2]
```diff
@@ -278,4 +279,8 @@ void sai_deserialize_queue_attr(
         _In_ const std::string& s,
         _Out_ sai_queue_attr_t& attr);
 
+void sai_deserialize_media_change_ntf(
```
Comment (lguohan) [inline meta/sai_serialize.h]: I am not sure if we have such ntf? can you point out the sai function for this?

File: meta/saiserialize.cpp (hunk 1 above)
Comment (dgsudharsan) [reply meta/saiserialize.cpp]: Hi Guohan
The data is already serialized in media_settings.json in json format. Now the task is to de-serialize the data from the json format to the port attributes(SAI_PORT_ATTR_SERDES_PREEMPHASIS, SAI_PORT_ATTR_SERDES_IDRIVER, SAI_PORT_ATTR_SERDES_IPREDRIVER)

File: meta/sai_serialize.h (hunk 2 above)
Comment (dgsudharsan) [reply meta/sai_serialize.h]: I think the name of the API might be little misleading here compared to other APIs present. The ntf stands for media notification. The intent of the API is to deserialize the notification received from xcvrd into list of SAI attributes. There are three attributes handled currently as mentioned in the above comment and in future many can be added. Thus i intended to keep the name of API as deserialize media notificaiton.
Please let me know if you want to rename the API

**Comment by stevenlu99** [comment]
The change looks ok for me.

File: meta/saiserialize.cpp (hunk 1 above)
Comment (lguohan) [reply meta/saiserialize.cpp]: I do not see how this SAI serialize is related to media_settings.json. they are parsed by xrcvd, send to orchagent app db, then go to sai.

**Comment by dgsudharsan** [comment]
> can you drop this PR? does not feel necessary.

Hi Guohan.. Sure. I am moving this functionality to sonic-swss
+    void sai_get_port_serdes_values (const std::string& s,
+                                                std::vector<uint32_t> &lane_values);

**Pull request closed without merging**